### ISSUE TITLE
iOS/iPadOS 16 issues

### ISSUE THREAD

**Issue by hucknz** [normen/homebridge-bravia#175]
Hi, 

Love your plug-in, being able to run remote free is great. 

I’ve been testing iOS and iPadOS 16 and come across a couple of issues. I’m not sure if they’re the plug-in or OS bugs. Happy to help with any testing you require. 

On iOS the volume control has stopped working. Pressing volume up/down changes the ringer. its still working fine on our iOS 15 devices. 

On iPadOS the device regularly disappears from the remote. Restarting seems to resolve the issue.

**Comment by normen** [comment]
I had the remote option disappear every now and then, otherwise I can't reproduce this. Usually the TV reappears in the remote app after a while (without rebooting anything). I put that down to a mdns/bonjour issue, no other idea really. Do you run the TV in external device mode or in "normal" mode?

**Comment by hucknz** [comment]
Yeah, it's a weird one. The overall setup with Homebridge and your plugin had been super consistent (no issues I can recall in over a year) until testing 16, hence my wondering if there's a bug with the plugin. I know they've made some quite big changes to Homekit in iOS 16 so I wondered if that was causing some sort of incompatibility. 

The iPad issue seems to clear up as soon as I restart the device. If I disconnect from wifi then rejoin it doesn't show up. 

Everything else (lights, etc.) seem to be working fine, though I did try add another Sony remote yesterday and that's not even showing up in the home app (but is in Homebridge) so maybe it's something affecting TV's in the new Homekit platform. 

The iOS problem is a little different. The TV always shows and I can control it (stop/start, etc.) but volume control has just randomly stopped working. Weirdly it started working again last night so I might just keep an eye on it and keep doing a bit more testing.

**Comment by benzman81** [comment]
I have these problems since the beginning of using it, so its not an iOS 16 Bug.

**Comment by normen** [comment]
Yeah, all of this to me looks like more hints at networking issues. It working or not randomly for some people, things being different when reconnecting etc.

**Comment by hucknz** [comment]
I can see where you guys are coming from. 

We've had this plugin running for a couple of years and I can't recall any issues with it over the last year, at least. It's been incredibly reliable so hats off to @normen for building something so solid. 

The reason I started here is that iOS 16 appeared to be the only commonality and this was the only plugin that has been having issues. iOS 15 devices didn't appear to be affected. 

I totally understand that the issue may not be caused by this plugin, it could well be an issue with Homebridge, or Homekit, but as it's only affecting our Sony TV I figured I'd ask here first then work my way down the layers. 

I got frustrated with my phone yesterday so I grabbed the iOS 15 device when I was having issues with changing the volume. Turns out the same issues are happening on my wife's phone, she just somehow didn't notice or her device continued working when mine didn't. 

I'll go ahead and close out the ticket as it seems like the root issue might lie somewhere else. Thanks for your help.
